lib/wayback_machine_downloader.rb

```diff
@@ -123,13 +124,53 @@ def get_file_list_curated
     end
     file_list_curated
   end
+  def get_file_list_ALL
+    file_list_curated = Hash.new
+    puts "snapshots_to_consider: " + get_all_snapshots_to_consider.lines.count.to_s
+    get_all_snapshots_to_consider.each_line do |line|
+      unless line.include?('/')
+        print "**" + line
+	next
+      end
+      file_timestamp = line[0..13].to_i
+      file_url = line[15..-2]
+      file_id = file_url.split('/')[3..-1].join('/')
+      file_id = [file_timestamp, file_id].join('/')
+      file_id = CGI::unescape file_id 
+      file_id = file_id.tidy_bytes unless file_id == ""
+      if file_id.nil?
+        puts "Malformed file url, ignoring: #{file_url}"
+      else
+        if match_exclude_filter(file_url)
+          puts "File url matches exclude filter, ignoring: #{file_url}"
+        elsif not match_only_filter(file_url)
+          puts "File url doesn't match only filter, ignoring: #{file_url}"
+        elsif file_list_curated[file_id]
+         #puts "__" + file_id
```

Comment:
I am working on this right now. It's actually working.

In this code `file_id` is not just the file id, but` file_id = [file_timestamp, file_id]`. I'll edit this part to make it clearer for futur readers. I was confused a bit too.